infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationServer.java

```diff
@@ -196,10 +218,17 @@ private void handleMetadataRequest(@Nonnull RequestMsg request,
 
             LogReplicationSinkManager sinkManager = sessionToSinkManagerMap.get(session);
 
+            // SinkManager not found can indicate a new session discovered or yet to be discovered.
+            // Drop the message if the incoming session is not known to sessionManger, otherwise create a corresponding
+            // sinkManager
             if (sinkManager == null) {
-                log.error("Sink Manager not found for session {}, total={}, sessions={}", session,
-                        sessionToSinkManagerMap.size(), sessionToSinkManagerMap.keySet());
-                return;
+                if(!sessionManager.getSessions().contains(session)) {
```

Comment:
We create a sink manager for any created session in the constructor. But we don't create a sink manager for a later created session (different behavior). Is this because these two events are async and you could receive a message from source before the sink manager is created? (still we don't cover the case where the session is ever known by the sink). If this is the case, I think it's worth a comment.